### PR TITLE
Acd 4119 schema validation for datamodel json files

### PR DESCRIPTION
**Pull request by harshshah2-crest**
- PR contains the code for schema validation of datamodel json files.
- User can also provide its own path for datamodels directory using `--dm-path`  parameter in pytest.
- Default value `pytest-splunk-addon\pytest_splunk_addon\standard_lib\datamodels`  which contains CIM datamodels.

- This type of exception will be displayed to the user in case of invalid json files.
image

### CODE REVIEW

File: pytest_splunk_addon/splunk.py [hunk 1]
```diff
@@ -101,6 +101,13 @@ def pytest_addoption(parser):
             " of 8.0.0. "
         ),
     )
+    group.addoption(
+        "--dm-path",
```
Comment (jayjoshi-crest) [inline pytest_splunk_addon/splunk.py:105]: Can we update the param name to `--splunk-dm-path`?

File: pytest_splunk_addon/splunk.py [hunk 2]
```diff
@@ -101,6 +101,13 @@ def pytest_addoption(parser):
...
+        action="store",
+        dest="dm_path",
+        default="pytest-splunk-addon\pytest_splunk_addon\standard_lib\datamodels",
+        help=("Path to the datamodels directory"),
```
Comment (jayjoshi-crest) [inline pytest_splunk_addon/splunk.py:109]: Add the detail that 
- relative or absoute path can be provided. 
- Json files are expected in the directory.
- Where can a user find the schema for those json.

File: pytest_splunk_addon/splunk.py (hunk 1 above)
Comment (harshshah2-crest) [reply pytest_splunk_addon/splunk.py]: Done.

File: pytest_splunk_addon/splunk.py (hunk 2 above)
Comment (harshshah2-crest) [reply pytest_splunk_addon/splunk.py]: Done.

**Pull request merged**